presto-main/src/main/java/com/facebook/presto/type/TypeRegistry.java

```diff
@@ -326,7 +331,73 @@ public static boolean canCoerce(Type actualType, Type expectedType)
     public static boolean canCoerce(TypeSignature actualType, TypeSignature expectedType)
     {
         Optional<TypeSignature> commonSuperTypeSignature = getCommonSuperTypeSignature(actualType, expectedType);
-        return commonSuperTypeSignature.isPresent() && commonSuperTypeSignature.get().equals(expectedType);
+        return commonSuperTypeSignature.isPresent() && typesMatch(commonSuperTypeSignature.get(), expectedType);
+    }
+
+    /**
+     * Checks type signatures for equality with the only difference:
+     * <p>
+     * If the long parameters are not specified in `expectedType` than any calculated long values will be accepted
+     * <p>
+     * e.g.
+     * <p>
+     * typesMatch(DECIMAL(2,1), DECIMAL) -> TRUE
```

Comment:
`DECIMAL` is syntactic sugar for `DECIMAL(MAX_PRECISION, 0)`, so why would this return true?